tests/neg-custom-args/deprecation/t3235-minimal.scala

```diff
@@ -1,8 +1,8 @@
 object Test {
   def main(args: Array[String]): Unit = {
-    assert(123456789.round == 123456789) // error
-    assert(math.round(123456789) == 123456789) // error
-    assert(1234567890123456789L.round == 1234567890123456789L) // error
-    assert(math.round(1234567890123456789L) == 1234567890123456789L) // error
+    assert(123456789.round == 123456789) // error (Int to Flaot)
+    assert(math.round(123456789) == 123456789)
```

Comment:
That behavior seems correct. `round` is overloaded, so there is no expected type yet. This seems to indicate that the tests are done in the wrong phase. Maybe we should test this only after Typer?

Reply:
Maybe test `assert(math.round(123456789) == 123456789)` in another test file to test the other error/warning.

Reply:
Yes, the test was correct.  I just wanted a pull request ready that we could merge today to unbreak the nightly builds.  I spent some time with Lukas and I think we should be able to merge #13911 instead.  If something comes up I'll PR moving the test instead.

Reply:
Ironically, @nicolasstucki, it's thanks to the fact that these were in the same file that we even found that the wrong warning was being emitted.  I'm still not sure I like this new "// error" with no message assertion practice that scala 3 has introduced..